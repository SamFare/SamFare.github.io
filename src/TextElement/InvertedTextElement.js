import React from "react";
import TextElement from "./TextElement.js";

import css from "./TextElement.css";

export default class InvertedTextElement extends TextElement{
  render(){
    return (
      <div className={css.textElement}>
        <div> 
          <div className={css.textHeadingInverted}> {this.props.title} </div> 
          <div className={css.textElement}>
            {this.props.text}
          </div> 
        </div>    
        <div className={css.subElement}>
          <img className={css.experenceImg} src={this.props.img}  height="100" width="100"/>
        </div> 
      </div>
    );
  }
}