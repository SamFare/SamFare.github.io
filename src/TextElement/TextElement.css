
.textElement {
    display: block;
    white-space:pre-wrap;
    max-width: 100vw;
    margin-bottom: 5px;
    min-height: 100px;
}

.textHeading {
    padding-top: 1vh;
    font-size: 24pt;
    font-family: 'Haettenschweiler', sans-serif;
    color: #555962;
}

.subElement { 
    margin: 10px;
    height: fit-content;
}

.experenceImg {
    border: 5px;
}

.textElementText {
    max-width: 100vw;
}

.leftAlign {
    text-align: left;
}

.rightAlign {
    text-align: right; 
}

@media only screen and (max-width: 600px) {
    .rightAlign {
        text-align: left; 
    }
}