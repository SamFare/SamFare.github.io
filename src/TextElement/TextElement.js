import React from "react";
import PropTypes from "prop-types";

import css from "./TextElement.css";

export default class TextElement extends React.Component{
  
  constructor(props) {
    super(props);
    this.textValue = this.props.text;
  }

  static get propTypes() { 
    return {
      text: PropTypes.string,
      img: PropTypes.string
    };
  }

  render(){
    return (
      <div className={css.textElement}>  
        <div className={css.subElement}>
          <img src={this.props.img}  height="100" width="100"/>
        </div>
        <div className={css.textElement}>
          {this.props.text}
        </div>  
      </div>
    );
  }
}