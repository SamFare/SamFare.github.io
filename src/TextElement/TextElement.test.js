import React from "react";
import TextElement from "./TextElement.js";
import { mount } from "enzyme";


test("When no orientation is selected, the image displays on the left", () => { 
  const wrapper = mount(<TextElement />);
  const img = wrapper.find(".qa-img");
  expect(img.prop("style"))
    .toHaveProperty("float", "left");
});

test("When left orientation is selected, the header text is left aligned", () => { 
  const wrapper = mount(<TextElement />);
  const headerText = wrapper.find(".qa-header-text");
  expect(headerText.prop("className")).toEqual(expect.stringContaining("leftAlign"));
});

test("When no orientation is selected, the header text is left aligned", () => { 
  const wrapper = mount(<TextElement orientation="left"/>);
  const headerText = wrapper.find(".qa-header-text");
  expect(headerText.prop("className")).toEqual(expect.stringContaining("leftAlign"));
});
  
test("When right orientation is selected, the image displays on the right", () => { 
  const wrapper = mount(<TextElement orientation="right"/>);
  const img = wrapper.find(".qa-img");
  expect(img.prop("style"))
    .toHaveProperty("float", "right");
});

test("When right orientation is selected, the header text is right aligned", () => { 
  const wrapper = mount(<TextElement orientation="right"/>);
  const headerText = wrapper.find(".qa-header-text");
  expect(headerText.prop("className")).toEqual(expect.stringContaining("rightAlign"));
});

test("When left is selected, the image displays on the left", () => { 
  const wrapper = mount(<TextElement orientation="left" />);
  const img = wrapper.find(".qa-img");
  expect(img.prop("style"))
    .toHaveProperty("float", "left");
});

test("When garbage is selected, the image displays on the left", () => { 
  const wrapper = mount(<TextElement orientation="meh" />);
  const img = wrapper.find(".qa-img");
  expect(img.prop("style"))
    .toHaveProperty("float", "left");
});

test("When no garbage is selected, the header text is left aligned", () => { 
  const wrapper = mount(<TextElement orientation="meh"/>);
  const headerText = wrapper.find(".qa-header-text");
  expect(headerText.prop("className")).toEqual(expect.stringContaining("leftAlign"));
});
  
