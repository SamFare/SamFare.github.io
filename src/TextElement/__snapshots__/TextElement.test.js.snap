// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A text element it has the image that was passed in 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <img
      className={undefined}
      height="100"
      src="test.png"
      width="100"
    />
  </div>
  <div>
    <div
      className={undefined}
    >
       
       
    </div>
    <div
      className={undefined}
    >
      text
    </div>
  </div>
</div>
`;
